mmaprototype: move unused helper functions to constraint_unused.go

wenyihu6 opened the pull request:
Epic: CRDB-55052 
Release note: none

---
**mmaprototype: move unused helper functions to constraint_unused.go**

Previously, constraint.go included a few helper functions intended for future
production use but not currently in use. This commit moves those helpers into
their own dedicated file to make constraint.go smaller.

---
**mmaprototype: move unused helper functions to constraint_unused.go**

Previously, constraint_matcher.go included a few helper functions intended for future
production use but not currently in use. This commit moves those helpers into
their own dedicated file to make constraint_matcher.go smaller.

wenyihu6 committed:
mmaprototype: move unused helper functions to constraint_unused.go

Previously, constraint.go included a few helper functions intended for future
production use but not currently in use. This commit moves those helpers into
their own dedicated file to make constraint.go smaller.

wenyihu6 committed:
mmaprototype: move unused helper functions to constraint_unused.go

Previously, constraint_matcher.go included a few helper functions intended for future
production use but not currently in use. This commit moves those helpers into
their own dedicated file to make constraint_matcher.go smaller.

sumeerbhola reviewed (commented):
So these will be moved back later? Is it just to make it easy to figure out what to unit test thoroughly now versus later?

> own dedicated file to make ... smaller

This doesn't sound like a good justification since it will become bigger later.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/157928#-:-OeIyLH06dW5qNNgxLRL:b-e1k48p)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

wenyihu6 reviewed (commented):
Yeah, this is just to make the code reading task of prototype smaller. We can move them back as they’re actually used.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/157928#-:-OeIz7nt--WsvjSZTx63:b-lo0edr)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (commented):
> Yeah, this is just to make the code reading task of prototype smaller. We can move them back as they’re actually used.

This doesn't sound like a strong enough justification to me, but I'll defer to @tbg.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/157928#-:-OeJ5Wzd-AbbkD1TD1CJ:b5xzmbz)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

tbg committed:
mmaprototype: move unused constraint functions to _test.go file

They are being used from tests. Make this clearer by
making these functions available only in the test.
If we decide to actually use them in production,
they can be moved back.

tbg commented:
^-- the bot found what looks like a pre-existing bug. Filed as https://github.com/cockroachdb/cockroach/issues/157991.

tbg reviewed (approved):
I can see where @wenyihu6's coming from. The fact that we have a bunch of functionality that isn't actually used is confusing, and makes it more difficult to see the "true" feature set of the current mmaprototype package. Since Wenyi, I, and perhaps Angela are all going to be reading this code, it makes sense to move anything that is "not truly there" out of the way to avoid confusion.

I tacked on a follow-up commit that moves the files into a `_test.go` file. These methods seem to be exercised, but only in tests.

I'll be interested in understanding whether when we expect to need this code and based on that determine whether to delete or resuscitate.

> @tbg reviewed 2 of 3 files at r1, 2 of 2 files at r2, 3 of 3 files at r3, 3 of 3 files at r4, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/157928#-:-OeLVOM_638zU9VadTaO:b-ydz0wk)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @wenyihu6)



<!-- Sent from Reviewable.io -->

tbg commented:
bors r+
